src/tests/end-to-end/tests/target-page/tabstop.test.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { ElementHandle } from 'puppeteer';
+
+import { Browser } from '../../common/browser';
+import { launchBrowser } from '../../common/browser-factory';
+import { fastPassSelectors } from '../../common/element-identifiers/fastpass-selectors';
+import { tabStopShadowDomSelectors, TargetPageElementSelectors } from '../../common/element-identifiers/target-page-selectors';
+import { PopupPage } from '../../common/page-controllers/popup-page';
+import { TargetPage } from '../../common/page-controllers/target-page';
+
+describe('tabstop tests', () => {
+    let browser: Browser;
+    let targetPage: TargetPage;
+    let popupPage: PopupPage;
+
+    beforeEach(async () => {
+        browser = await launchBrowser({ suppressFirstTimeDialog: true });
+        targetPage = await browser.newTargetPage({ testResourcePath: 'native-widgets/input-type-radio.html' });
+    });
+
+    afterEach(async () => {
+        if (browser) {
+            await browser.close();
+        }
+    });
+
+    test('works when tabstop is triggered from adhoc panel', async () => {
+        popupPage = await browser.newPopupPage(targetPage);
+        await popupPage.gotoAdhocPanel();
+        await popupPage.enableToggleByAriaLabel('Tab stops');
+
+        await targetPage.bringToFront();
+        await targetPage.waitForSelector(TargetPageElementSelectors.targetPageNativeWidgetFirstRadio);
```

Comment:
Is this really necessary? This doesn't perform any interaction with the selector, it just ensures it's present on the page.

Reply:
This is basically waiting for the selector to be in focus. I can change it to a `click` and it will mostly increase the reliability of the test.

Reply:
Removed. The test still works fine. Thanks for making me try and remove this. 👍

Reply:
Are you sure? That doesn't match [the description of `page.waitForSelector` in the Puppeteer API docs](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagewaitforselectorselector-options); it doesn't specify that it scrolls to the element or focuses the element, it just waits for it to be present in the DOM. I think this is probably a no-op as is and I would be surprised if adding a click changed the reliability of the test.